netlogo-gui/src/main/editor/CodeCompletionPopup.scala

```diff
@@ -0,0 +1,135 @@
+  val scrollPane = new JScrollPane(suggestionDisplaylist)
+  val window = new JDialog()
+  window.setUndecorated(true)
+  window.add(scrollPane)
+  window.setSize(new Dimension(150, 200))
+
+  scrollPane.setHorizontalScrollBarPolicy(ScrollPaneConstants.HORIZONTAL_SCROLLBAR_AS_NEEDED)
+  scrollPane.setVerticalScrollBarPolicy(ScrollPaneConstants.VERTICAL_SCROLLBAR_ALWAYS)
+
+  def init(editorArea: EditorArea): Unit = {
+    this.editorArea = editorArea
+    suggestionDisplaylist.setSelectionMode(ListSelectionModel.SINGLE_INTERVAL_SELECTION)
+    suggestionDisplaylist.setLayoutOrientation(JList.VERTICAL)
+    suggestionDisplaylist.setVisibleRowCount(10)
+    suggestionDisplaylist.setCellRenderer(new SuggestionListRenderer(editorArea))
+
+    suggestionDisplaylist.addKeyListener(new KeyListener {
+      override def keyTyped(e: KeyEvent): Unit = {
+        e.getKeyChar match {
+          case java.awt.event.KeyEvent.VK_ESCAPE =>
+            isPopupEnabled = false
+            window.setVisible(false)
+          case java.awt.event.KeyEvent.VK_BACK_SPACE =>
+            // Trying to fix the weird behaviour of backspace
+            val position = editorArea.getCaretPosition
+            if (position != 0) {
+              editorArea.setCaretPosition(position - 1)
+              val doc = editorArea.getDocument.asInstanceOf[javax.swing.text.PlainDocument]
+              doc.remove(position - 1, 1)
+            }
+          case java.awt.event.KeyEvent.VK_ENTER =>
+            val suggestion = suggestionDisplaylist.getSelectedValue
+            lastSuggested = suggestion
+            val token = compiler.getTokenAtPosition(editorArea.getText(), editorArea.getCaretPosition)
+            val position = editorArea.getCaretPosition
+            val doc = editorArea.getDocument.asInstanceOf[javax.swing.text.PlainDocument]
+            doc.replace(token.start, position - token.start, suggestion, null)
+            editorArea.setCaretPosition(token.start + suggestion.length)
+            isPopupEnabled = false
+          case _ =>
+            e.setSource(editorArea)
+            editorArea.dispatchEvent(e)
+        }
+      }
+
+      override def keyPressed(e: KeyEvent): Unit = {}
+
+      override def keyReleased(e: KeyEvent): Unit = {}
+    })
+  }
+
+  /**
+    * Makes the suggestion box initially and displays it.
+    * Should only be called when the user presses the shortcut to
+    * enable the suggestion. To update the suggestions list call showPopUp()
+    */
+  def enablePopup(): Unit = {
+    isPopupEnabled = true
+    showPopup()
+    val position = editorArea.getCaretPosition
+    window.setLocation(editorArea.getLocationOnScreen.x + editorArea.modelToView(position + 1).x,
+      editorArea.getLocationOnScreen.y + editorArea.modelToView(position).y + editorArea.getFont.getSize)
+  }
+
+  /**
+    * Updates the list of the suggestion box and makes it visible.
+    */
+  def showPopup(): Unit = {
```

Comment:
Both `enablePopup` and `showPopup` have comments explaining what they do and their name is not part of that comment. I would like to see them named a bit differently to indicate their functions - perhaps `displayPopup` and `updatePopup`?